tests/src/tests/cryptarchia/happy.rs

```diff
@@ -31,7 +31,10 @@ async fn happy_test(topology: &Topology) {
         .floor() as u64;
     let timeout = adjust_timeout(Duration::from_secs(timeout));
     let timeout = tokio::time::sleep(timeout);
-    #[expect(clippy::redundant_pub_crate)]
+    #[expect(
+        clippy::redundant_pub_crate,
+        reason = "Generated by `tokio::select` macro."
+    )]
```

Comment:
This is actually bothersome. It feels it really pollutes the code. I guess at some point that will be fixed by tokio guys 🤔

Reply:
Yes I think so too. An alternative approach would be to declare our own macro that wraps around this, and have the `expect` in a single place. What do you think?

Reply:
There has been discussion going on, and it seems the issue [has been fixed and merged recently](https://github.com/rust-lang/rust-clippy/pull/13952), so I would expect it to be gone in one of the upcoming Rust updates. In the meanwhile, I think it would still be ok to refactor the macro by wrapping it into another macro that will ignore the lint.

Reply:
Ok it does not seem to be possible, at the moment, and as I said in another comment, it seems to have been fixed in an upcoming Clippy version, so I would suggest we leave it as it is for now, as bothersome as it might seem.